src/rgw/rgw_admin.cc

```diff
@@ -4811,6 +4828,50 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT_BUCKETS_LIMIT_CHECK) {
+    void *handle;
+    std::list<std::string> user_ids;
+    metadata_key = "user";
+    int max = 1000;
+    int ret = store->meta_mgr->list_keys_init(metadata_key, &handle);
+    if (ret < 0) {
+      cerr << "ERROR: buckets limit check can't get user metadata_key: "
+	   << cpp_strerror(-ret) << std::endl;
+      return -ret;
+    }
+
+    bool truncated;
+
+    if (! user_id.empty()) {
+      user_ids.push_back(user_id.id);
+      ret =
+	RGWBucketAdminOp::limit_check(store, bucket_op, user_ids, f,
+	  warnings_only);
+    } else {
+      /* list users in groups of max-keys, then perform user-bucket
+       * limit-check on each group */
+      do {
+	ret = store->meta_mgr->list_keys_next(handle, max, user_ids,
+					      &truncated);
+	if (ret < 0 && ret != -ENOENT) {
+	  cerr << "ERROR: buckets limit check lists_keys_next(): "
+	       << cpp_strerror(-ret) << std::endl;
+	  return -ret;
```

Comment:
an early return here will leak memory allocated by `list_keys_init()`. `break;` should work instead